core/lib/common/utils.js

```diff
@@ -29,14 +29,53 @@ const exec = Bluebird.promisify(require('child_process').exec);
 const { fs } = require('mz');
 const keygen = Bluebird.promisify(require('ssh-keygen'));
 const path = require('path');
+const SSH = require('node-ssh');
+const assignIn = require('lodash/assignIn');
+
+
+const getSSHClientDisposer = (config) => {
+	const createSSHClient = (conf) => {
+		return Bluebird.resolve(
+			new SSH().connect(
```

Comment:
This creates a new SSH connection per command, which is unnecessary overhead. We should be able to open a connection and multiplex commands through it until we're done, or the connection is broken and we need to reconnect. This won't make a big difference locally, but could save a lot of time when the round-trip latency is high.

Also, I understand that this is a copy of how SSH connections are currently done, so it's not worse than the current approach, but I wanted to point it out.